dropshot/src/websocket.rs

```diff
@@ -51,11 +51,14 @@ pub type WebsocketEndpointResult = Result<Response<Body>, HttpError>;
 pub struct WebsocketConnection(WebsocketConnectionRaw);
 
 /// A type that implements [tokio::io::AsyncRead] + [tokio::io::AsyncWrite].
-pub type WebsocketConnectionRaw = hyper::upgrade::Upgraded;
+// Not public, since it uses less-stable hyper-util.
+type WebsocketConnectionRaw = hyper_util::rt::TokioIo<hyper::upgrade::Upgraded>;
```

Comment:
I don't know enough about all consumers to know if being able to name the type is important. I left it as unnamed type since it means less code, and if people need it named, it's not breaking to add a name to it like you said. 🤷

Reply:
I understand it to be a bit of an anti-pattern to have `pub` types without names because it makes it harder for consumers to work with them (e.g., what if you want to store it into a struct, or accept it as an argument to a function)?  If it's just a question of more code, I'd like to do the newtype.  That will also avoid one of the breaking changes here, right?

Reply:
FWIW, I wouldn't call it an anti-pattern. It's a pretty helpful feature, I use it as a library author specifically in places where the traits matter, but the exact type is not important. Most people just need to "use" the value in a function. If someone (rarely) needs to store it, usually they can box it. If it's very common to store, I do give it a name.

But I've added it back here.

Reply:
Thanks!  Yeah, I can see the appeal.  I just think it's hard to know what a consumer will (or won't) want to do -- I've been on the other end of that where the author assumed I wouldn't want to store a value and I was just stuck.

@jgallagher pointed out that there has been some [discussion](https://github.com/rust-lang/api-guidelines/discussions/60) on official guidance but apparently no decision.